Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/app/app.component.ts

```diff
@@ -9,24 +10,46 @@ import { Component, OnInit } from '@angular/core';
 export class AppComponent {
 
   public input: any;
-  title = 'Telemetry Recruit Training';
-  weather = 50;
-  speed = 0;
-  battery = 0;
+  title: string;
+  weather: number;
+  speed: number;
+  battery: number;
+  range: number;
+  speedInput: FormControl;
+  batteryInput: FormControl;
+  outputMessage: string;
+  buttonPress: boolean;
+
+  constructor() {
+    this.title = 'Telemetry Recruit Training';
+    this.weather = 50;
+    this.weather = 0;
+    this.speed = 0;
+    this.battery = 0;
+    this.speedInput = new FormControl(null, [Validators.required, Validators.min(0), Validators.max(90)]);
+    this.batteryInput = new FormControl(null, [Validators.required, Validators.min(0), Validators.max(100)]);
+    this.outputMessage = "";
```

Comment:
We use ' for strings instead of ". Running ng lint will tell you about any formatted conventions.